### DISCUSSION TITLE
[Script request] Authentik

### DISCUSSION THREAD

**Discussion by Feramance** [community-scripts/ProxmoxVE#53] [Request script]
### Application Name

Authentik

### Website

https://goauthentik.io/

### Description

Authentik is an IdP (Identity Provider) and SSO (single sign on) that is built with security at the forefront of every piece of code, every feature, with an emphasis on flexibility and versatility.

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by Aloe-recite** [comment]
Add my vote, this seems a very well-known app and probably deserves a place here.

**Reply by judeibe** [reply]
If somebody is up to a script can be submitted through a pr. That would quicken how fast it can be added.

**Reply by remz1337** [reply]
I might give it a shot if more people show their interest. There's a bit of reverse engineering that needs to be done with the Dockerfile, but looking at it briefly, it doesn't look too bad

**Reply by VenomousRage** [reply]
I'd like to add my vote for an Authentik script being added. I've been futzing around with Authentik and Traefik with their individual docker compose(s?) for two months.

**Reply by MickLesk** [reply]
@remz1337 If you do an Script, please set category to "in Pipeline" 🙂

**Comment by VenomousRage** [comment]
I'm not sure if it helps in any way, but I came across this (now archived) discussion from ttek's original github: https://github.com/tteck/Proxmox/discussions/2952 that started the process, as best I can tell without my morning coffee.

**Reply by remz1337** [reply]
That's great, definitely a good starting point. Thanks for pointing it out. I'll get working on this

**Comment by remz1337** [comment]
I got a first draft working. Can someone try it out and let me know if all the features are working?
`bash -c "$(wget -qLO - https://github.com/remz1337/ProxmoxVE/raw/remz/ct/authentik.sh)"`

**Reply by rzumbado** [reply]
First, let me thank you beforehand. You are a life saver, this is exactly what I was looking for (authentik on lxc).

Now, to the issue at hand, I get this error when running your script:
![image](https://github.com/user-attachments/assets/f9051e69-0fc8-4005-b0ec-fb78dc432571)

**Reply by VenomousRage** [reply]
I've had that (or a similar) error with other proxmox scripts. I've rerun it with verbose mobe enabled, to watch when the error shows up, as well as rerun it again without verbose mode. More often than not, in my experience, it boil(s/ed) down to some sort of dependency not being pulled/installed in the exact order that the script would like.

My "solution" was to rerun the script, as the "dependency" in question would have been installed during the first run.

**Reply by rzumbado** [reply]
I manually started "installing" everything just as the installation script does. I found out that the npm install fails due to some chrome driver or something missing. So I did npm install --force and it worked after that. This is the line I had to modify:
![image](https://github.com/user-attachments/assets/fc76fd9e-e53b-4553-a70a-6ecc59023330)

After that change, it installed all dependencies and built successfully. I will keep going and see if I manage to install it this way. Just an FYI in case you needed to know a "work around" for now.

**Reply by rzumbado** [reply]
well you can ignore me now... I figure the problem was the RAM usage.. when installing this I used 1GB of RAM which I thought was 10GBs... that was the cause of the problem. Sorry for the confusion.

**Comment by VenomousRage** [comment]
32:09 minutes (Optiplex 7060 8500T using all 6 cores and 8GB of RAM), but the first draft of the script installed. There were a few warnings and error, but I'm logged into the UI. Now, to play with setting it up.

**Reply by remz1337** [reply]
Awesome, very glad to hear. Indeed the installation takes some time as it needs to build Python and and the app itself. It does take around 30 minutes on my server too.

Please report back once you have tested all the features are working. After that, I will open a PR

Thanks

**Reply by MickLesk** [reply]
Heavy resources. It will be interesting to see whether the script, when it is finalized, causes more problems than it makes sense.

**Reply by remz1337** [reply]
Feel free to have a look and see if there's any potential optimizations. I'm all in for a faster build :)

**Comment by VenomousRage** [comment]
So, a few days later... still as solid as the day I installed it. No issues at all in functionality or setting up my usual configs.

**Reply by tesla-76** [reply]
Container up for more than 24 hours, no problems so far. Takes some patience to get it compiled but ultimately completes everything without any problems.

**Reply by MickLesk** [reply]
Okay, that sounds good. We'll just revise the script and then test it in the project.

**Reply by bvdberg01** [reply]
Same here, testing is now for a few days and works great!

**Reply by remz1337** [reply]
awesome, opening a PR now. Thank you all for testing the script

**Reply by Aloe-recite** [reply]
Guys, any update on this? :) 
Thanks!

**Comment by second2falcon** [comment]
Take my vote for an authentik script. Sadly @remz1337 's doesn't seem to work on my install.

I am always getting a `/dev/fd/63: line 43: exit: You need to set 'PHS_ADD_SSH_USER' variable.: numeric argument required` error.

Any ideas on how to fix it? Tried with customized settings and with default.

**Reply by remz1337** [reply]
Hey @second2falcon sorry, yea I need to fix that bug in my fork. It's not related to Authentik per se, I'm working on something else. Anyway, the PR is almost done, so if you are patient a bit, it should be merged into the official repo soon
https://github.com/community-scripts/ProxmoxVE/pull/291

**Comment by ayoahha** [comment]
Hi,
Is it normal that the install is taking soooo long, at the step:

 ⠋ Setting up Python 3 
 
 (Maybe more than 10 minutes from beg)

---
i was looking to the bash script, and before opening an issue, i just want to validate.
From a screenshot:

![image](https://github.com/user-attachments/assets/3da210a9-650a-44e1-80f0-e72283b66b35)

> Is there a typo on the line:
```bash
$STD update-alternatives --install /usr/bin/python3 python3 /usr/local/bin/python3.12 1
```
There is a space between 12 and 1.

Cheers and thanks for this script
